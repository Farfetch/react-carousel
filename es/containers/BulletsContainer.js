function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { Bullets } from "../components";
import { CarouselContext } from "../context";
import React from 'react';

var BulletsContainer = function BulletsContainer(props) {
  return /*#__PURE__*/React.createElement(CarouselContext.Consumer, null, function (_ref) {
    var activeItem = _ref.activeItem,
        itemsLength = _ref.itemsLength,
        isRTL = _ref.isRTL,
        itemsToScroll = _ref.itemsToScroll;
    return /*#__PURE__*/React.createElement(Bullets, _extends({}, props, {
      activeItem: activeItem,
      itemsLength: itemsLength,
      isRTL: isRTL,
      itemsToScroll: itemsToScroll
    }));
  });
};

export default BulletsContainer;