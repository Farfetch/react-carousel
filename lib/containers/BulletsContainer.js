"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _components = require("../components");

var _context = require("../context");

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var BulletsContainer = function BulletsContainer(props) {
  return /*#__PURE__*/_react["default"].createElement(_context.CarouselContext.Consumer, null, function (_ref) {
    var activeItem = _ref.activeItem,
        itemsLength = _ref.itemsLength,
        isRTL = _ref.isRTL,
        itemsToScroll = _ref.itemsToScroll;
    return /*#__PURE__*/_react["default"].createElement(_components.Bullets, _extends({}, props, {
      activeItem: activeItem,
      itemsLength: itemsLength,
      isRTL: isRTL,
      itemsToScroll: itemsToScroll
    }));
  });
};

var _default = BulletsContainer;
exports["default"] = _default;
module.exports = exports.default;