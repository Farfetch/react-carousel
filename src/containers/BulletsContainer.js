import { BaseBullets } from '#base';
import { CarouselContext } from '#context';
import React from 'react';

const BulletsContainer = (props) => (
    <CarouselContext.Consumer>
        {({ activeItem, itemsToShow, itemsLength, isRTL, itemsToScroll }) => (
            <BaseBullets
                {...props}
                itemsToShow={itemsToShow}
                activeItem={activeItem}
                itemsLength={itemsLength}
                isRTL={isRTL}
                itemsToScroll={itemsToScroll}
            />
        )}
    </CarouselContext.Consumer>
);

export default BulletsContainer;
