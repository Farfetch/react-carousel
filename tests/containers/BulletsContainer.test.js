import { createSerializer } from 'enzyme-to-json';
import { shallow } from 'enzyme';
import BulletsContainer from '../../src/containers/BulletsContainer';
import React from 'react';

const mockContextCarousel = {
    activeItem: 1,
    itemsLength: 10,
    containerRef: {
        current: null,
    },
};

beforeAll(() => {
    expect.addSnapshotSerializer(createSerializer());
});

describe('<BulletsContainer/>', () => {
    it('should render correctly', () => {
        const tree = shallow(<BulletsContainer />).renderProp('children')({
            ...mockContextCarousel,
        });

        expect(tree).toMatchSnapshot();
    });

    it('should render correctly with custom Props', () => {
        const mockTheme = {
            container: 'container',
        };
        const tree = shallow(
            <BulletsContainer data-testId="test" theme={mockTheme} />
        ).renderProp('children')({
            ...mockContextCarousel,
        });

        expect(tree.find('Bullets').prop('data-testId')).toBe('test');
        expect(tree.find('Bullets').prop('theme')).toEqual(mockTheme);
    });

    it('should render correctly with correct Context', () => {
        const tree = shallow(<BulletsContainer />).renderProp('children')({
            ...mockContextCarousel,
        });

        expect(tree.find('Bullets').prop('activeItem')).toBe(
            mockContextCarousel.activeItem
        );
        expect(tree.find('Bullets').prop('itemsLength')).toBe(
            mockContextCarousel.itemsLength
        );
    });
});
