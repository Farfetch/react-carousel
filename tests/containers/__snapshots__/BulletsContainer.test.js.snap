// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BulletsContainer/> should render correctly 1`] = `
<Bullets
  activeItem={1}
  itemsLength={10}
  itemsToScroll={1}
/>
`;
